### PR TITLE
Fix/libre explorer

### PR DESCRIPTION
**Pull request by mfshao**
A little more fixes to data explorer in libre mode.

If TIER_ACCESS_LEVEL is `regular`, default filter will be `with-access`, otherwise, `all-data` will be the default value.
Putting above logic into constructor since the logic in `getSnapshotBeforeUpdate` doesn't get triggered always.

Tested locally against icgc (libre), my dev env (regular) and qa-ibd (private) and all seems to be working fine.

### Bug Fixes
- Ensure data explorer filter defaults to `with-access` only if TIER_ACCESS_LEVEL is `regular`, otherwise set it to `all-data`
- Fixed logic for explorer component locks: don't lock if TIER_ACCESS_LEVEL is `libre`
- Fixed `Each child in a list should have a unique “key” prop` warning for ExplorerVisualization

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -12139,24 +12140,25 @@
       "dependencies": {
         "abbrev": {
           "version": "1.1.1",
-          "resolved": false,
+          "resolved": "https://registry.npmjs.org/abbrev/-/abbrev-1.1.1.tgz",
```
Comment (ZakirG) [inline package-lock.json]: are u sure about this stuff?

Comment (mfshao) [reply package-lock.json]: `npm i` just gives me these...

File: src/GuppyDataExplorer/TierAccessSelector/index.jsx [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
- * By default "All Data" is selected
+ * By default 'Data with Access' is selected if TIER_ACCESS_LEVEL is 'regular'
+ * Otherwise 'Data with Access' is selected
  */
 class TierAccessSelector extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      selected: 'with-access',
+      selected: (tierAccessLevel === 'regular') ? 'with-access' : 'all-data',
```
[the remark this answers is not in the document]
Comment (mfshao) [reply src/GuppyDataExplorer/TierAccessSelector/index.jsx]: > is this wanted? I think only regular level use this UI and other levels won't need this anyways...

indeed, it is not needed

File: src/GuppyDataExplorer/ExplorerVisualization/index.jsx [hunk 3]
```diff
@@ -99,7 +99,8 @@ class ExplorerVisualization extends React.Component {
     const chartData = this.getData(this.props.aggsData, this.props.chartConfig, this.props.filter);
     const tableColumns = (this.props.tableConfig.fields && this.props.tableConfig.fields.length > 0)
       ? this.props.tableConfig.fields : this.props.allFields;
-    const isComponentLocked = checkForAnySelectedUnaccessibleField(this.props.aggsData,
+    // don't lock components for libre commons
+    const isComponentLocked = (tierAccessLevel === 'libre') ? false : checkForAnySelectedUnaccessibleField(this.props.aggsData,
```
[the remark this answers is not in the document]
Comment (mfshao) [reply src/GuppyDataExplorer/ExplorerVisualization/index.jsx]: good call!

**Pull request merged**